site/packages/aa-signers/turnkey/introduction.md

```diff
@@ -0,0 +1,95 @@
+---
+outline: deep
+head:
+  - - meta
+    - property: og:title
+      content: TurnkeySigner
+  - - meta
+    - name: description
+      content: Overview of the TurnkeySigner class in aa-signers
+  - - meta
+    - property: og:description
+      content: Overview of the TurnkeySigner class in aa-signers
+---
+
+# Turnkey Signer
+
+`TurnkeySigner` is a signer implementation which extends `SmartAccountAuthenticator` to leverage the [Turnkey SDK](https://docs.turnkey.com/category/sdk). It supports features such as authentication, message and typed data signing, and authentication details retrieval.
+
+`TurnkeySigner` provides implementations for all methods on `SmartAccountAuthenticator`:
+
+1.  [`authenticate`](/packages/aa-signers/turnkey/authenticate) -- supports user authentication.
+2.  [`getAddress`](/packages/aa-signers/turnkey/getAddress) -- supports typed data signatures from the smart contract account's owner address.
+3.  [`signMessage`](/packages/aa-signers/turnkey/signMessage) -- supports message signatures.
+4.  [`signTypedData`](/packages/aa-signers/turnkey/signTypedData) -- supports typed data signatures.
+5.  [`getAuthDetails`](/packages/aa-signers/turnkey/getAuthDetails) -- supports authentication details retrieval.
+
+## Install Dependencies
+
+`TurnkeySigner` requires installation of the [`@turnkey/http`](https://github.com/tkhq/sdk/tree/main/packages/http) and [`@turnkey/viem`](https://github.com/tkhq/sdk/tree/main/packages/viem) dependencies. `aa-signers` lists them as optional dependencies.
+
+Every request to Turnkey must be signed using a [stamper](https://docs.turnkey.com/category/api-design). In this example, we use the API key stamper from [`@turnkey/api-key-stamper`](https://github.com/tkhq/sdk/tree/main/packages/api-key-stamper).
+
+::: code-group
+
+```bash [npm]
+npm i -s @turnkey/api-key-stamper
+npm i -s @turnkey/http
+npm i -s @turnkey/viem
+```
+
+```bash [yarn]
+yarn add @turnkey/api-key-stamper
+yarn add @turnkey/http
+yarn add @turnkey/viem
+```
+
+## Usage
+
+::: code-group
+
+```ts [example.ts]
+import { TurnkeySigner } from "@alchemy/aa-signers";
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+
+const turnkeySigner = new TurnkeySigner({
+  apiUrl: "api.turnkey.com",
+  stamper: new ApiKeyStamper({
+    apiPublicKey: "TURNKEY_API_PUBLIC_KEY",
+    apiPrivateKey: "TURNKEY_API_PRIVATE_KEY",
+  }),
+});
+
+const authParams = {
+  organizationId: "TURNKEY_ORGANIZATION_ID",
+  signWith: "TURNKEY_WALLET_ID",
```

Comment:
the example doesn't create the wallet, so i agree that it might be helpful to show how the dev can obtain the wallet id/address f they don't store that for users. since the wallet id/address is only needed for the `authenticate` call and this alternative example is a bit lengthy, wdyt about me adding it [here](https://github.com/alchemyplatform/aa-sdk/pull/292/files#diff-fdf51fa5129147a3ef7bf55922343105cd7e6f7d4995e147c019e81662794ca7R25)?

Reply:
yea that sounds good to me